### DISCUSSION TITLE
Why we need runtime ops?

### DISCUSSION THREAD

**Discussion by janetat** [tinygrad/tinygrad#6222] [Q&A] [answered]
# Question
1. In https://github.com/tinygrad/tinygrad/tree/master/tinygrad/runtime folder, why we need ops_*.py?
2. What's the point of https://github.com/tinygrad/tinygrad/blob/master/tinygrad/runtime/ops_nv.py?

**Comment by mikex86** [comment, marked as answer]
It's a complete replacement of the cuda userspace responsible for launching cuda kernels.
The point is to not be dependent on it, similar to `ops_amd.py`, where it isn't just nice to have, but necessary for stability.
